modules/ROOT/pages/mock-file-cookbook.adoc

```diff
@@ -0,0 +1,68 @@
+= Mocking Resources for Your Tests
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+When the tests suites start to grow in size and complexity, it’s a good practice to externalize some parts of the code to test resources so that you can keep the code clean and organized as well as reuse the files that you need to mock for your tests without the need to have duplicated code.
+
+In Mule 3, you used to use a `getResourceAsString()` function to incorporate external or mocked resources to your tests. In Mule 4, you can leverage DataWeave to mock your code for your tests.
+
+== Mocking Files Using DataWeave Scripts
+
+You can use a DataWeave script that returns the data that you want to mock, save the file under a folder in `src/test/resources` and then use a DataWeave function to read this file.
+
+For example, define a `mockPost.dwl` file in the `src/test/resources/sample_data` folder with the following content:
+
+[source]
+----
+%dw 2.0
+output application/json
+---
+{
+	"foo" : "var"
+}
+----
+
+This DataWeave file is a sample that creates a mock payload that you want to return in an `http:request`. When configuring the `then-return` section in the `mock-when`, use the `readUrl` function to read the mapping file:
+
+[source,xml,linenums]
+----
+<munit-tools:then-return>
+  <munit-tools:payload value="#[readUrl('classpath://sample_data/mockPost.dwl')]" mediaType="application/json" encoding="UTF-8" />
+</munit-tools:then-return>
+----
+
+The `classpath-based` URL uses the `classpath:` protocol prefix to find the file we defined under `src/test/resources`.
+
+== Mocking Files Using Variables in a DataWeave Module
+
+You can create a DataWeave module to centralize all your mocks in a single file by declaring them in DataWeave variables.
```

Comment:
Here and at the beginning of the script text, consider adding a bit about WHY a user might choose using scripts vs. variables in modules.

Reply:
```suggestion
You can create a DataWeave module to centralize all your mocked data in a single file by declaring it using DataWeave variables.
```